### DISCUSSION TITLE
Force redirect to login page

### DISCUSSION THREAD

**Discussion by noxify** [blitz-js/blitz#2533] [Q&A] [answered]
Hi,

i ran into an issue and I'm not sure if this is an bug or just user handling 😏

## What's the goal?

I have a page which requires that the user needs to be logged in.
If the user isn't logged in, the user should be redirected to the login page.

## What did I to do this?

I used the `app/pages/index.tsx` for testing.

I added the following after `Home.getLayout = ...`:

```
Home.authenticate = {
  redirectTo: '/login'
}
```

I expected that I will be redirected to the login page and will see the login form.
But the current result is the following:

* URL in to browser has been changed from `http://localhost:3000/` to `http://localhost:3000/login?next=%2F`
* The page title has been changed from `Home` to `Login`
* The content has NOT been changed

In my test instance ( just a fresh blitz app via `blitz new blitztest` ), there is no error in the web console.

## The one million dollar question

* Is this the expected behavior?
** Yes? Is there a way to do a force reload?
** No? Did I missed something?

**Comment by KiritoXF** [comment, marked as answer]
I think you should just use `Home.authenticate = true;`  rather than redirect to other page.
In my case, it will replace page content to login but the url won't change.
If you use redirect, it will show Home page shortly and redirect to login page. Howerver, guest user will see the page content unexpected.

**Reply by noxify** [reply]
@KiritoXF this seems to work - Thanks.

> In my case, it will replace page content to login but the url won't change.

I can confirm this - But that's ok for me.

> If you use redirect, it will show Home page shortly and redirect to login page. Howerver, guest user will see the page content unexpected.

I have this behavior also with `Home.authenticate = true`.


https://user-images.githubusercontent.com/521777/123793407-f1266a80-d8e1-11eb-808d-ba26a1f251aa.mp4

tested it also w/o changing anything, but the behavior is the same.

Possible solution:
Removing `<Suspense>` Part from `app/pages/_app.tsx` solves the problem, but I'm not sure what the side effects are.

**Reply by noxify** [reply]
Without `<Suspense>` also ` Home.authenticate = { redirectTo: '/login' }` is working as expected.

**Reply by flybayer** [reply]
@noxify I think there's a bug in react causing that. I've removed `<Suspense>` from the new app template until that is fixed: https://github.com/blitz-js/blitz/pull/2548
